homeassistant/components/reolink/update.py

```diff
@@ -0,0 +1,78 @@
+"""Update entities for Reolink devices."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from reolink_aio.exceptions import ReolinkError
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkBaseCoordinatorEntity
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up update entities for Reolink component."""
+    reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([ReolinkUpdateEntity(reolink_data)])
+
+
+class ReolinkUpdateEntity(ReolinkBaseCoordinatorEntity, UpdateEntity):
+    """Update entity for a Netgear device."""
+
+    _attr_device_class = UpdateDeviceClass.FIRMWARE
+    _attr_supported_features = UpdateEntityFeature.INSTALL
+    _attr_release_url = "https://reolink.com/download-center/"
+    _attr_name = "Update"
+
+    def __init__(
+        self,
+        reolink_data: ReolinkData,
+    ) -> None:
+        """Initialize a Netgear device."""
+        super().__init__(reolink_data, reolink_data.firmware_coordinator)
+
+        self._attr_unique_id = f"{self._host.unique_id}_update"
```

Comment:
We don't need to add the platform domain to the unique_id. The entity registry is aware of both the integration domain and the platform domain. It's good to add a unique suffix if we're planning to add more than one entity type for the platform.